## 📝 Chap07-2. Stocks with GPT

#### yfinance 란?
- 야후 파이낸스의 금융 데이터를 쉽게 가져올 수 있게 해주는 오픈소스 파이썬 라이브러리.
- 주가, 재무재표, 거래량 등 다양한 데이터를 데이터프레임 형태로 가져올 수 있다.

In [1]:
import yfinance as yf
from rich.pretty import pprint

# Microsoft (MSFT)에 대한 Ticker 객체 생성
msft = yf.Ticker("MSFT")

# Ticker 객체에 대한 정보 출력 (.py에서 실행할 때는 print(msft.info)로 사용)
pprint(msft.info)

{
│   'address1': 'One Microsoft Way',
│   'city': 'Redmond',
│   'state': 'WA',
│   'zip': '98052-6399',
│   'country': 'United States',
│   'phone': '425 882 8080',
│   'website': 'https://www.microsoft.com',
│   'industry': 'Software - Infrastructure',
│   'industryKey': 'software-infrastructure',
│   'industryDisp': 'Software - Infrastructure',
│   'sector': 'Technology',
│   'sectorKey': 'technology',
│   'sectorDisp': 'Technology',
│   'longBusinessSummary': "Microsoft Corporation, a technology company, develops and supports a portfolio of technology solutions for individuals and businesses worldwide. Its products include operating systems, server applications, business solution applications, software development tools, desktop and server management tools, and video games; and devices, such as PCs, tablets, gaming and entertainment consoles, other intelligent devices, and related accessories. The company's Productivity and Business Processes segment offers Microsoft 365 Commercial, Enterprise Mobility + Security, Power BI, Exchange, SharePoint, Microsoft Teams, Microsoft 365 Security and Compliance, Microsoft 365 Copilot, and Windows Commercial on-premises and Office licensed on-premises. This segment also provides Microsoft 365 Consumer products and cloud services; LinkedIn, including talent solutions, marketing solutions, subscriptions, and sales solutions; Dynamics 365, a set of cloud-based applications; and on-premises ERP and CRM applications. Its Intelligent Cloud segment offers server products and cloud services; cloud and AI consumption-based services, GitHub cloud services, health and life sciences cloud services, as well as virtual desktop offerings, and other cloud services; SQL Server, Windows Server, Visual Studio, System Center, and related client access licenses; and enterprise support services, industry solutions, Microsoft partner network, and learning experience. The company's More Personal Computing segment provides Windows OEM licensing and devices comprising Surface and PC accessories; XBOX hardware, and XBOX first- and third-party content and services; XBOX Game Pass and other subscriptions; XBOX Cloud Gaming, advertising, and other cloud services; and search advertising consisting of Bing, Copilot, Microsoft News, Microsoft Edge, and third-party affiliates. It sells its products through partners and retail networks. The company was founded in 1975 and is headquartered in Redmond, Washington.",
│   'fullTimeEmployees': 223000,
│   'companyOfficers': [
│   │   {
│   │   │   'maxAge': 1,
│   │   │   'name': 'Mr. Satya  Nadella',
│   │   │   'age': 58,
│   │   │   'title': 'Chairman & CEO',
│   │   │   'yearBorn': 1967,
│   │   │   'fiscalYear': 2025,
│   │   │   'totalPay': 12251294,
│   │   │   'exercisedValue': 0,
│   │   │   'unexercisedValue': 0
│   │   },
│   │   {
│   │   │   'maxAge': 1,
│   │   │   'name': 'Mr. Bradford L. Smith LCA',
│   │   │   'age': 66,
│   │   │   'title': 'President & Vice Chairman',
│   │   │   'yearBorn': 1959,
│   │   │   'fiscalYear': 2025,
│   │   │   'totalPay': 4532750,
│   │   │   'exercisedValue': 0,
│   │   │   'unexercisedValue': 0
│   │   },
│   │   {
│   │   │   'maxAge': 1,
│   │   │   'name': 'Ms. Amy E. Hood',
│   │   │   'age': 53,
│   │   │   'title': 'Executive VP & CFO',
│   │   │   'yearBorn': 1972,
│   │   │   'fiscalYear': 2025,
│   │   │   'totalPay': 4444191,
│   │   │   'exercisedValue': 0,
│   │   │   'unexercisedValue': 0
│   │   },
│   │   {
│   │   │   'maxAge': 1,
│   │   │   'name': 'Mr. Takeshi  Numoto',
│   │   │   'age': 54,
│   │   │   'title': 'Executive VP & Chief Marketing Officer',
│   │   │   'yearBorn': 1971,
│   │   │   'fiscalYear': 2025,
│   │   │   'totalPay': 2871250,
│   │   │   'exercisedValue': 0,
│   │   │   'unexercisedValue': 0
│   │   },
│   │   {
│   │   │   'maxAge': 1,
│   │   │   'name': 'Mr. Judson B. Althoff',
│   │   │   'age': 52,
│   │   │   'title': 'Executive VP & CEO of Commercial Business',
│   │   │   'y

##### 죄근 5일간의 주가 정보 확인

In [6]:
hist = msft.history(period="5d") # 5일간의 주가 데이터를 가져옴
display(hist) # 데이터 출력

,Open,High,Low,Close,Volume,Dividends,Stock Splits
Date,,,,,,,
2026-08-24 00:00:00-04:00,483.209991,490.609985,481.859985,487.309998,17230800,0.0,0.0
2026-08-25 00:00:00-04:00,485.440002,492.440002,484.299988,491.709991,19519000,0.0,0.0
2026-08-26 00:00:00-04:00,487.850006,497.399994,487.309998,496.369995,20758900,0.0,0.0
2026-08-27 00:00:00-04:00,494.880005,506.480011,490.079987,505.059998,28688800,0.0,0.0
2026-08-28 00:00:00-04:00,NaN,NaN,NaN,NaN,28953449,0.0,0.0


In [3]:
msft.recommendations # 추천 정보 출력

,period,strongBuy,buy,hold,sell,strongSell
0,0m,14,38,3,0,0
1,-1m,14,40,3,0,0
2,-2m,13,41,3,0,0
3,-3m,12,41,3,0,0


- period : 분석가들이 추천한 시점
- 추천은 strongBuy ~ strongSell 까지 5개의 등급으로 분류

#### Streamlit 에서 코드 실행하기

(chap07) PS C:\Users\ksol9\Documents\ksol927_ai_study\StudyRepo\learn_doit_llm_agent\chap07> streamlit run what_time_is_it_streamlit.py
2026-08-29 16:31:31.565 Uvicorn server started on :::8501

  You can now view your Streamlit app in your browser.

  Local URL: http://localhost:8501
  Network URL: http://211.250.198.194:8501

  Help agents write better Streamlit apps?
  Install the official Streamlit skills by running streamlit skills in your terminal.

ChatCompletionMessage(content=None, refusal=None, role='assistant', annotations=[], audio=None, function_call=None, tool_calls=[ChatCompletionMessageFunctionToolCall(id='call_fg3FAqJ1dzJUunDBShwqzgMC', function=Function(arguments='{"ticker":"TSLA"}', name='get_yf_stock_info'), type='function')])
{'address1': '1 Tesla Road', 'city': 'Austin', 'state': 'TX', 'zip': '78725', 'country': 'United States', 'phone': '512 516 8177', 'website': 'https://www.tesla.com', 'industry': 'Auto Manufacturers', 'industryKey': 'auto-manufacturers', 'industryDisp': 'Auto Manufacturers', 'sector': 'Consumer Cyclical', 'sectorKey': 'consumer-cyclical', 'sectorDisp': 'Consumer Cyclical', 'longBusinessSummary': 'Tesla, Inc. designs, develops, manufactures, leases, and sells electric vehicles, and energy generation and storage systems in the United States, China, and internationally. The company operates in two segments, Automotive; and Energy Generation and Storage. The company offers electric vehicles, as well as sells automotive regulatory credits; and non-warranty maintenance services and collision, automotive insurance services, as well as part sales and retail merchandise sale. It also provides sedans and sport utility vehicles through direct and used vehicle sales, a network of Tesla Superchargers, and in-app upgrades; purchase financing and leasing services; services for electric vehicles through its company-owned service locations and Tesla mobile service technicians; and vehicle limited warranties and extended service plans. In addition, the company engages in the design, manufacture, installation, sale, and leasing of solar energy generation and energy storage products, and related services to residential, commercial, and industrial customers and utilities through its website, stores, and galleries, as well as through a network of channel partners. Further, it provides services and repairs to its energy product customers, including under warranty and extended service plans; and various financing options to its residential customers; lithium-ion battery energy storage products, such as Powerwall and Megapack; energy generation products, including solar panels and solar roof; self-driving development and artificial intelligence software, vehicle control and infotainment software, and battery and powertrain. The company was formerly known as Tesla Motors, Inc. and changed its name to Tesla, Inc. in February 2017. Tesla, Inc. was incorporated in 2003 and is headquartered in Austin, Texas.', 'fullTimeEmployees': 134785, 'companyOfficers': [{'maxAge': 1, 'name': 'Mr. Elon R. Musk', 'age': 54, 'title': 'Co-Founder, Technoking of Tesla, CEO & Director', 'yearBorn': 1971, 'fiscalYear': 2025, 'exercisedValue': 0, 'unexercisedValue': 129602732032}, {'maxAge': 1, 'name': 'Mr. Vaibhav  Taneja', 'age': 47, 'title': 'Chief Financial Officer', 'yearBorn': 1978, 'fiscalYear': 2025, 'totalPay': 403000, 'exercisedValue': 15033838, 'unexercisedValue': 417696224}, {'maxAge': 1, 'name': 'Mr. Xiaotong  Zhu', 'age': 45, 'title': 'Senior Vice President of APAC & Global Vehicle Manufacturing', 'yearBorn': 1980, 'fiscalYear': 2025, 'totalPay': 376322, 'exercisedValue': 4548562, 'unexercisedValue': 808327104}, {'maxAge': 1, 'name': 'Travis  Axelrod', 'title': 'Head of Investor Relations', 'fiscalYear': 2025, 'exercisedValue': 0, 'unexercisedValue': 0}, {'maxAge': 1, 'name': 'Mr. Brandon  Ehrhart', 'title': 'General Counsel & Corporate Secretary', 'fiscalYear': 2025, 'exercisedValue': 0, 'unexercisedValue': 0}, {'maxAge': 1, 'name': 'Mr. Franz  von Holzhausen', 'title': 'Chief Designer', 'fiscalYear': 2025, 'exercisedValue': 0, 'unexercisedValue': 0}, {'maxAge': 1, 'name': 'Ms. Roxanne  Inskip-Kaye', 'title': 'Head of Operations - Electric Vehicle Business', 'fiscalYear': 2025, 'exercisedValue': 0, 'unexercisedValue': 0}, {'maxAge': 1, 'name': 'Mr. John  Walker', 'age': 62, 'title': 'Vice President of Sales - North America', 'yearBorn': 1963, 'fiscalYear': 2025, 'totalPay': 121550, 'exercisedValue': 0, 'unexercisedValue': 0}, {'maxAge': 1, 'name': 'Mr. Rodney D. Westmoreland Jr.', 'title': 'Director of Construction Management', 'fiscalYear': 2025, 'exercisedValue': 0, 'unexercisedValue': 0}, {'maxAge': 1, 'name': 'Mr. Lars  Moravy', 'title': 'Vice President of Vehicle Engineering', 'fiscalYear': 2025, 'exercisedValue': 0, 'unexercisedValue': 0}], 'auditRisk': 4, 'boardRisk': 10, 'compensationRisk': 10, 'shareHolderRightsRisk': 9, 'overallRisk': 10, 'governanceEpochDate': 1785542400, 'compensationAsOfEpochDate': 1767139200, 'executiveTeam': [], 'maxAge': 86400, 'priceHint': 2, 'previousClose': 354.81, 'open': 357.095, 'dayLow': 345.2, 'dayHigh': 358.8, 'regularMarketPreviousClose': 354.81, 'regularMarketOpen': 357.095, 'regularMarketDayLow': 345.2, 'regularMarketDayHigh': 358.8, 'payoutRatio': 0.0, 'beta': 1.827, 'trailingPE': 329.00946, 'forwardPE': 161.56454, 'volume': 32818355, 'regularMarketVolume': 32818355, 'averageVolume': 41413253, 'averageVolume10days': 35391090, 'averageDailyVolume10Day': 35391090, 'bid': 345.22, 'ask': 348.0, 'bidSize': 1, 'askSize': 1, 'marketCap': 1377404715008, 'nonDilutedMarketCap': 1401338910865, 'fiftyTwoWeekLow': 297.38, 'fiftyTwoWeekHigh': 498.83, 'allTimeHigh': 498.83, 'allTimeLow': 0.998667, 'fulldayPrice': 348.19, 'fulldayChange': -6.06, 'fulldayChangePercent': -1.70796, 'priceToSalesTrailing12Months': 13.292974, 'fiftyDayAverage': 361.4264, 'twoHundredDayAverage': 401.41074, 'trailingAnnualDividendRate': 0.0, 'trailingAnnualDividendYield': 0.0, 'currency': 'USD', 'tradeable': False, 'enterpriseValue': 1374555865088, 'profitMargins': 0.03671, 'floatShares': 2819186930, 'sharesOutstanding': 3949547394, 'sharesShort': 69196896, 'sharesShortPriorMonth': 70646375, 'sharesShortPreviousMonthDate': 1784073600, 'dateShortInterest': 1786665600, 'sharesPercentSharesOut': 0.0175, 'heldPercentInsiders': 0.17819001, 'heldPercentInstitutions': 0.43702, 'shortRatio': 1.82, 'shortPercentOfFloat': 0.0196, 'impliedSharesOutstanding': 3949547394, 'bookValue': 21.995, 'priceToBook': 15.855876, 'lastFiscalYearEnd': 1767139200, 'nextFiscalYearEnd': 1798675200, 'mostRecentQuarter': 1782777600, 'earningsQuarterlyGrowth': -0.049, 'netIncomeToCommon': 3806000128, 'trailingEps': 1.06, 'forwardEps': 2.15858, 'pegRatio': 5.13, 'lastSplitFactor': '3:1', 'lastSplitDate': 1661385600, 'enterpriseToRevenue': 13.265, 'enterpriseToEbitda': 127.806, '52WeekChange': 0.077271104, 'SandP52WeekChange': 0.1966995, 'quoteType': 'EQUITY', 'currentPrice': 348.75, 'targetHighPrice': 600.0, 'targetLowPrice': 125.0, 'targetMeanPrice': 390.09283, 'targetMedianPrice': 410.0, 'recommendationMean': 2.34783, 'recommendationKey': 'buy', 'numberOfAnalystOpinions': 39, 'totalCash': 43524001792, 'totalCashPerShare': 11.02, 'ebitda': 10755000320, 'totalDebt': 16080000000, 'quickRatio': 1.347, 'currentRatio': 1.941, 'totalRevenue': 103619002368, 'debtToEquity': 18.373, 'revenuePerShare': 32.055, 'returnOnAssets': 0.019299999, 'returnOnEquity': 0.046669997, 'grossProfits': 19534000128, 'freeCashflow': 4835999744, 'operatingCashflow': 18684999680, 'earningsGrowth': -0.03, 'revenueGrowth': 0.255, 'grossMargins': 0.18852, 'ebitdaMargins': 0.10379, 'operatingMargins': 0.014099999, 'financialCurrency': 'USD', 'symbol': 'TSLA', 'language': 'en-US', 'region': 'US', 'typeDisp': 'Equity', 'quoteSourceName': 'Nasdaq Real Time Price', 'triggerable': True, 'customPriceAlertConfidence': 'HIGH', 'regularMarketChangePercent': -1.70796, 'regularMarketPrice': 348.75, 'corporateActions': [], 'postMarketTime': 1787961599, 'regularMarketTime': 1787947200, 'marketState': 'CLOSED', 'shortName': 'Tesla, Inc.', 'longName': 'Tesla, Inc.', 'exchange': 'NMS', 'messageBoardId': 'finmb_27444752', 'exchangeTimezoneName': 'America/New_York', 'exchangeTimezoneShortName': 'EDT', 'gmtOffSetMilliseconds': -14400000, 'market': 'us_market', 'esgPopulated': False, 'hasPrePostMarketData': True, 'firstTradeDateMilliseconds': 1277818200000, 'postMarketChangePercent': -0.16057278, 'postMarketPrice': 348.19, 'postMarketChange': -0.55999756, 'regularMarketChange': -6.06, 'regularMarketDayRange': '345.2 - 358.8', 'fullExchangeName': 'NasdaqGS', 'averageDailyVolume3Month': 41413253, 'fiftyTwoWeekLowChange': 51.369995, 'fiftyTwoWeekLowChangePercent': 0.17274192, 'fiftyTwoWeekRange': '297.38 - 498.83', 'fiftyTwoWeekHighChange': -150.07999, 'fiftyTwoWeekHighChangePercent': -0.300864, 'fiftyTwoWeekChangePercent': 7.7271104, 'earningsTimestamp': 1784750400, 'earningsTimestampStart': 1792612800, 'earningsTimestampEnd': 1792612800, 'earningsCallTimestampStart': 1784755800, 'earningsCallTimestampEnd': 1784755800, 'isEarningsDateEstimate': True, 'epsTrailingTwelveMonths': 1.06, 'epsForward': 2.15858, 'epsCurrentYear': 1.77267, 'priceEpsCurrentYear': 196.73712, 'fiftyDayAverageChange': -12.676392, 'fiftyDayAverageChangePercent': -0.035073232, 'twoHundredDayAverageChange': -52.660736, 'twoHundredDayAverageChangePercent': -0.13118915, 'sourceInterval': 15, 'exchangeDataDelayedBy': 0, 'averageAnalystRating': '2.3 - Buy', 'cryptoTradeable': False, 'displayName': 'Tesla', 'trailingPegRatio': 5.1987}
AI      : 테슬라, Inc.(Tesla, Inc.)는 다양한 국가에서 전기차와 에너지 생성 및 저장 시스템을 설계, 개발, 제조, 임대 및 판매하는 회사입니다. 회사는 자동차 부문과 에너지 생성 및 저장 부문으로 나뉘어 운영됩니다.

주요 정보:
- 산업: 자동차 제조
- 본사 위치: 텍사스주 오스틴
- 설립 연도: 2003
- CEO: 엘론 머스크
- 직원 수: 약 134,785명

테슬라는 다양한 전기차 모델을 제공하며, 자동차와 관련된 서비스와 제품(규제 크레딧, 유지보수 서비스, 자동차 보험 등)도 제공합니다. 또한, 자사의 웹사이트와 오프라인 스토어를 통해 태양광 에너지 생성 및 저장 제품(솔라 패널, 솔라 루프, 파워월 및 메가팩)을 디자인, 제조, 설치, 판매 및 임대합니다.

재무 및 시장 정보:
- 현재 주가: $348.75
- 시장 자본: 약 $1.38조
- 기업가치: $1.37조
- 수익성: 연간 순이익률 약 3.67%
- 52주 최고가: $498.83
- 52주 최저가: $297.38

테슬라는 기술 및 인공지능 소프트웨어에도 투자하여 자율 주행 개발을 진행 중이며, 에너지 관련 제품에 대한 다양한 재무 옵션도 제공합니다. 더불어 전기차 서비스는 자사 소유 서비스 센터와 모바일 서비스 기술자를 통해 지원됩니다.

더 궁금한 사항이 있으면 언제든 물어보세요!

ChatCompletionMessage(content=None, refusal=None, role='assistant', annotations=[], audio=None, function_call=None, tool_calls=[ChatCompletionMessageFunctionToolCall(id='call_ewNjCNgXi6DeXiIZrPAENL2g', function=Function(arguments='{"timezone":"America/Chicago"}', name='get_current_time'), type='function')])
2026-08-29 02:35:17 America/Chicago
AI      : 현재 테슬라 본사가 위치한 미국 텍사스주 오스틴의 시간은 2026년 8월 29일 오전 2시 35분입니다. 시차 정보 등의 변화가 있을 수 있으니 참고하세요!

#### 최근 주가기록과 추천정보 결과

ChatCompletionMessage(content=None, refusal=None, role='assistant', annotations=[], audio=None, function_call=None, tool_calls=[ChatCompletionMessageFunctionToolCall(id='call_99IDEbjGtae4dgxrGD2c6dtD', function=Function(arguments='{"ticker":"MSFT","period":"1mo"}', name='get_yf_stock_history'), type='function')])
| Date                      |    Open |    High |     Low |   Close |      Volume |   Dividends |   Stock Splits |
|:--------------------------|--------:|--------:|--------:|--------:|------------:|------------:|---------------:|
| 2026-07-29 00:00:00-04:00 | 392.661 | 400.496 | 388.01  | 389.806 | 4.7209e+07  |        0    |              0 |
| 2026-07-30 00:00:00-04:00 | 437.077 | 457.828 | 431.627 | 450.252 | 1.10161e+08 |        0    |              0 |
| 2026-07-31 00:00:00-04:00 | 449.154 | 465.963 | 448.486 | 463.847 | 6.0846e+07  |        0    |              0 |
| 2026-08-03 00:00:00-04:00 | 475.235 | 490.726 | 474.107 | 486.734 | 6.67621e+07 |        0    |              0 |
| 2026-08-04 00:00:00-04:00 | 479.996 | 498.502 | 478.27  | 491.884 | 5.04555e+07 |        0    |              0 |
| 2026-08-05 00:00:00-04:00 | 495.427 | 497.304 | 484.767 | 486.544 | 3.33889e+07 |        0    |              0 |
| 2026-08-06 00:00:00-04:00 | 487.632 | 500.618 | 487.602 | 498.921 | 3.64415e+07 |        0    |              0 |
| 2026-08-07 00:00:00-04:00 | 498.272 | 504.231 | 497.793 | 499.051 | 2.88462e+07 |        0    |              0 |
| 2026-08-10 00:00:00-04:00 | 502.494 | 512.765 | 501.326 | 505.109 | 3.12061e+07 |        0    |              0 |
| 2026-08-11 00:00:00-04:00 | 503.382 | 504.38  | 498.611 | 502.863 | 2.30498e+07 |        0    |              0 |
| 2026-08-12 00:00:00-04:00 | 499.051 | 500.558 | 490.596 | 491.505 | 2.90015e+07 |        0    |              0 |
| 2026-08-13 00:00:00-04:00 | 492.343 | 500.398 | 492.084 | 495.946 | 2.3039e+07  |        0    |              0 |
| 2026-08-14 00:00:00-04:00 | 495.427 | 499.071 | 492.992 | 494.469 | 1.62176e+07 |        0    |              0 |
| 2026-08-17 00:00:00-04:00 | 489.219 | 491.734 | 477.511 | 479.447 | 2.96358e+07 |        0    |              0 |
| 2026-08-18 00:00:00-04:00 | 480.635 | 483.36  | 476.253 | 480.725 | 2.40876e+07 |        0    |              0 |
| 2026-08-19 00:00:00-04:00 | 479.168 | 488.381 | 478.459 | 483.4   | 1.99619e+07 |        0    |              0 |
| 2026-08-20 00:00:00-04:00 | 483.3   | 484.16  | 479.5   | 481.15  | 2.00208e+07 |        0.91 |              0 |
| 2026-08-21 00:00:00-04:00 | 479.88  | 486.36  | 478.53  | 483.24  | 2.25108e+07 |        0    |              0 |
| 2026-08-24 00:00:00-04:00 | 483.21  | 490.61  | 481.86  | 487.31  | 1.72308e+07 |        0    |              0 |
| 2026-08-25 00:00:00-04:00 | 485.44  | 492.44  | 484.3   | 491.71  | 1.9519e+07  |        0    |              0 |
| 2026-08-26 00:00:00-04:00 | 487.85  | 497.4   | 487.31  | 496.37  | 2.07589e+07 |        0    |              0 |
| 2026-08-27 00:00:00-04:00 | 494.88  | 506.48  | 490.08  | 505.06  | 2.86888e+07 |        0    |              0 |
| 2026-08-28 00:00:00-04:00 | nan     | nan     | nan     | nan     | 2.89534e+07 |        0    |              0 |
AI      : 마이크로소프트 주식의 한 달 전 닫힘 가격은 약 389.81였고, 최근 닫힘 가격은 약 505.06입니다. 따라서, 한 달 전과 비교하여 현재 마이크로소프트 주식은 상승한 상태입니다.
ChatCompletionMessage(content=None, refusal=None, role='assistant', annotations=[], audio=None, function_call=None, tool_calls=[ChatCompletionMessageFunctionToolCall(id='call_cb8DwIDMHjKG6dxpAkGcVQJL', function=Function(arguments='{"ticker":"MSFT"}', name='get_yf_stock_recommendations'), type='function')])
|    | period   |   strongBuy |   buy |   hold |   sell |   strongSell |
|---:|:---------|------------:|------:|-------:|-------:|-------------:|
|  0 | 0m       |          14 |    38 |      3 |      0 |            0 |
|  1 | -1m      |          14 |    40 |      3 |      0 |            0 |
|  2 | -2m      |          13 |    41 |      3 |      0 |            0 |
|  3 | -3m      |          12 |    41 |      3 |      0 |            0 |
AI      : 마이크로소프트 주식에 대한 현재 애널리스트들의 추천을 보면, "Strong Buy" 14명, "Buy" 38명, "Hold" 3명으로, "Sell"이나 "Strong Sell" 추천은 없습니다. 이는 대체로 마이크로소프트 주식을 매수하는 것이 긍정적으로 평가되고 있다는 것을 의미합니다. 그러나, 최종 결정은 투자 목적과 재정 상황을 고려하여 신중하게 판단하시는 것이 좋습니다.